Sources/ParseSwift/Synchronous.swift

```diff
@@ -8,22 +8,22 @@
 
 import Foundation
 
-typealias ResultCapturing<T> = (Result<T>) -> Void
+typealias BlockCapturing<T> = (T?, Error?) -> Void
 // Mark it private for now
-private func await<T>(block: (@escaping ResultCapturing<T>) -> Void) throws -> T {
+private func await<T>(block: (@escaping BlockCapturing<T>) -> Void) throws -> T {
     let sema = DispatchSemaphore(value: 0)
-    var r: Result<T>!
+    var error: Error?
+    var value: T?
     block({
-        r = $0
-        sema.signal()
+        error = $1
+        value = $0
```

Comment:
Can we inverse those and add back sema.signal()?

Reply:
Just to clarify, it should read:
```swift
block({
    value = $0
    error = $1
    sema.signal()
})
```
Correct?

Reply:
Yep!

Reply:
@flovilmart Got it.